### PR TITLE
Issue 2232 gui testcases

### PR DESCRIPTION
**Pull request by maryvictol**
This PR adds testcases for NAT Gateway configuration #2232.

### CODE REVIEW

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+| 6 | Input `test` into the ***Server name:*** field and click outside the field | |
+| 7 | Mark ***Specify IP address*** checkbox | <li> **IP** field and **Resolve** button appear on the ***Add new route*** pop-up <li> Error message `Unable to resolve the given hostname: test` appears <li> Warnings `Field is required` appear under the ***IP*** and ***Port*** fields|
+| 8 | Clear the ***Server name:*** field, input `google.com` into it and click outside the field | Correct IP address appears in the ***IP*** field |
+| 9 | Clear the ***IP*** field, input `127.1.1` and click outside the field | Warning `Invalid format` appears under the ***IP*** field |
+| 10 | Click **Resolve** button | Correct IP address appears in the ***IP*** field |
+| 11 | Input `80` into the ***Port:*** field | **Add** button becomes active |
+| 12 | Input `port1` into the ***Comment:*** field | |
+| 13 | Click **Add** button | <li> A new record with the name specified at step 7 appears in the External Resources list <li> **Revert** and **Save** buttons become active |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md]: Fix `step 7` to `step 8`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+| 14 | Click **Revert** button | The record added at step 12 is removed from the External Resources list <li> **Revert** and **Save** buttons become disabled |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md]: Fix `step 12` to `step 13`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `google.com` into the ***Server name:*** field and click outside the field | | 
+| 7 | Mark ***Specify IP address*** checkbox | Resolved IP address appears in the ***IP*** field |
+| 8 | Input `80` into the ***Port:*** field | |
+| 9 | Input `port1` into the ***Comment:*** field | |
+| 10 | Click **Add** button | A new record with the name specified at step 6 appears in the External Resources list |
+| 11 | Click **Save** button | <li> Icon with status ***CREATION SCHEDULED*** appears for record added at step 9 <li> **Revert** and **Save** buttons become disabled |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Fix `step 9` to `step 10`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+| 12 | Wait 1 minute and click **Refresh** button | The record added at step 9 has <li> Icon with status ***ACTIVE*** <li> Internal *Service name* `<prefix>-google-com`, where `<prefix>` is NAT proxy service name prefix <li> Internal *IP* and *Port* <li> Comment added at step 8 |
+| 13 | Store *Internal IP* for the record added at step 9 | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Fix `step 9` to `step 10`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `google.com` into the ***Server name:*** field and click outside the field | | 
+| 7 | Mark ***Specify IP address*** checkbox | Resolved IP address appears in the ***IP*** field |
+| 8 | Input `80` into the ***Port:*** field | |
+| 9 | Input `port1` into the ***Comment:*** field | |
+| 10 | Click **Add** button | A new record with the name specified at step 6 appears in the External Resources list |
+| 11 | Click **Save** button | <li> Icon with status ***CREATION SCHEDULED*** appears for record added at step 9 <li> **Revert** and **Save** buttons become disabled |
+| 12 | Wait 1 minute and click **Refresh** button | The record added at step 9 has <li> Icon with status ***ACTIVE*** <li> Internal *Service name* `<prefix>-google-com`, where `<prefix>` is NAT proxy service name prefix <li> Internal *IP* and *Port* <li> Comment added at step 8 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Fix:
- `step 9` to `step 10`
- `step 8` to `step 9`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+| 13 | Store *Internal IP* for the record added at step 9 | | 
+| 14 | Open the **Tools** page | | 
+| 15 | Select test tool | |
+| 16 | Launch a tool with default settings | |
+| 17 | At the **Runs** page, click the just-launched run | | 
+| 18 | Wait until the **SSH** hyperlink appears | |
+| 19 | Click the **SSH** hyperlink | |
+| 20 | In the opened tab, enter and perform the command: <br>`unset http_proxy https_proxy` <br> `curl google.com -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to google.com <internal_ip> port <port>`, <br> where `<internal_ip>` is *Internal IP* stored at step 12, `<port>` is *port* specified at step 7 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Fix:
- `step 12` to `step 13`
- `step 7` to `step 8`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the admin user from the prerequisites | |
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `google.com` into the ***Server name:*** field and click outside the field | | 
+| 7 | Mark ***Specify IP address*** checkbox | Resolved IP address appears in the ***IP*** field |
+| 8 | Input `80` into the ***Port:*** field | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md:22]: Suggest to add a verification here - whether the gateway with the same server name and port already exists. And if yes - try to use another port in this case, for example

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `yahoo.com` into the ***Server name:*** field | | 
+| 7 | Input `80` into the ***Port:*** field | |
+| 8 | Input `port1` into the ***Comment:*** field | |
+| 9 | Click **Add** button | A new record with the name specified at step 6 appears in the External Resources list <li> external IP for this record is the same as name |
+| 10 | Click **Save** button | <li> Icon with status ***CREATION SCHEDULED*** appears for record added at step 9 <li> **Revert** and **Save** buttons become disabled |
+| 11 | Wait 1 minute and click **Refresh** button | The record added at step 9 has <li> Icon with status ***ACTIVE*** <li> Internal *Service name* `<prefix>-google-com`, where `<prefix>` is NAT proxy service name prefix <li> Internal *IP* and *Port* <li> Comment added at step 8 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md]: Fix `<prefix>-google-com` to `<prefix>-yahoo-com`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the admin user from the prerequisites | |
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `yahoo.com` into the ***Server name:*** field | | 
+| 7 | Input `80` into the ***Port:*** field | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md:22]: Suggest to add a verification here - whether the gateway with the same server name and port already exists. And if yes - try to use another port in this case, for example

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_4.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+| :---: | --- | --- |
+| 1 | Login as the admin user from the prerequisites | |
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Store *External IP* for the route added at case [2232_2](2232_2.md) | |
+| 6 | Click **Add Route** button | |
+| 7 | Input `google.com` into the ***Server name:*** field |  |
+| 8 | Mark ***Specify IP address*** checkbox | ***IP*** field with resolved IP address appears |
+| 9 | Clear the ***IP*** field and input value other than value stored at step 6 | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_4.md]: Fix `step 6` to `step 5`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_4.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+| 10 | Input `443` into the ***Port:*** field | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_4.md:20]: Suggest to add a verification here - whether the gateway with the same server name and port already exists. And if yes - try to use another port in this case, for example

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_5.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the admin user from the prerequisites | |
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `google.com` into the ***Server name:*** field |  |
+| 7 | Input `443` into the ***Port:*** field | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_5.md:17]: Suggest to add a verification here - whether the gateway with the same server name and port already exists. And if yes - try to use another port in this case, for example

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_6.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the admin user from the prerequisites | |
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `yahoo.com` into the ***Server name:*** field |  |
+| 7 | Mark ***Specify IP address*** checkbox |  |
+| 8 | Input `443` into the ***Port:*** field | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_6.md:18]: Suggest to add a verification here - whether the gateway with the same server name and port already exists. And if yes - try to use another port in this case, for example

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_7.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+| 9 | Mark ***Specify IP address*** checkbox | |
+| 10 | Clear the ***IP*** field and input value stored at step 5 | |
+| 11 | Input `80` into the ***Port:*** field | Warning `Value should be unique` appears under the ***Port:*** field |
+| 12 | Clear the ***Port*** field and input `443` | |
+| 13 | Click **Add** button | |
+| 14 | Click **Save** button | Icon with status ***CREATION SCHEDULED*** appears for record added at steps 13 |
+| 15 | Wait 1 minute and click **Refresh** button | The record added at step 13 has <li> icon with status ***ACTIVE*** <li> ***Internal IP*** that is the same as the value stored at step 6 |
+| 16 |At the **Runs** page, click the run launched at step 16 of [2232_2](2232_2.md) case| |
+| 17 | Click the **SSH** hyperlink | |
+| 18 | In the opened tab, enter and perform the command: <br>`unset http_proxy https_proxy` <br> `curl google.com:443 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to yahoo.com <internal_ip> port <port>`, <br> where `<internal_ip>` is *Internal IP* stored at step 6, `<port>` is *port* specified at step 12 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_7.md]: Fix `yahoo.com` to `google.com`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `bing.com` into the ***Server name:*** field |  |
+| 7 | Mark ***Specify IP address*** checkbox | |
+| 8 | Input `80` into the ***Port:*** field | |
+| 9 | Input `port1` into the ***Comment:*** field | |
+| 10 | Click **Add** button | |
+| 11 | Repeat steps 5 - 10 with values <li>`443` for the ***Port:*** field <li> `port2` for the ***Comment:*** field | |
+| 12 | Click **Save** button | Icons with status ***CREATION SCHEDULED*** appear for records added at steps 5 and 10 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md]: Fix `steps 5 and 10` to `steps 10 and 11`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+| 13 | Wait 1 minute and click **Refresh** button | The records added at steps 5 and 10 have <li> icons with status ***ACTIVE*** <li> the same *Internal IPs*, different *Internal Ports* and *Comments* |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md]: Fix `steps 5 and 10` to `steps 10 and 11`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+| 14 | Store *Internal IP* for the route saved at step 12 | |
+| 15 | At the **Runs** page, click the run launched at step 16 of [2232_2](2232_2.md) case| |
+| 16 | Click the **SSH** hyperlink | |
+| 17 | In the opened tab, enter and perform the command: <br>`unset http_proxy https_proxy` <br> `curl bing.com:80 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to bing.com <internal_ip> port 80`, <br> where `<internal_ip>` is *Internal IP* stored at step 15 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md]: Fix `step 15` to `step 14`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+| 18 | In the opened tab, enter and perform the command: <br> `curl bing.com:443 -v -ipv4` | The command output contains: <br> `Trying <internal_ip>...>` <br> `Connected to bing.com <internal_ip> port 443`, <br> where `<internal_ip>` is *Internal IP* stored at step 15 |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md]: Fix `step 15` to `step 14`

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the admin user from the prerequisites | |
+| 2 | Open the **Settings** page | |
+| 3 | Click the **SYSTEM MANAGEMENT** tab | |
+| 4 | Click the **NAT GATEWAY** tab | |
+| 5 | Click **Add Route** button | |
+| 6 | Input `bing.com` into the ***Server name:*** field |  |
+| 7 | Mark ***Specify IP address*** checkbox | |
+| 8 | Input `80` into the ***Port:*** field | |
```
Comment (NShaforostov) [inline docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_10.md:23]: Suggest to add a verification here - whether the gateway with the same server name and port already exists. And if yes - try to use another port in this case, for example

**Review by NShaforostov — COMMENTED** [review]
Fix minor notes, please.
See comments for details.

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md (hunk 1 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md (hunk 2 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_1.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md (hunk 3 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md (hunk 4 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md (hunk 5 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md (hunk 6 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_2.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md (hunk 8 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md (hunk 9 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_3.md:22]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_4.md (hunk 10 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_4.md]: Done

File: docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_7.md (hunk 14 above)
Comment (maryvictol) [reply docs/testcases/UI/System_Management/2232_NAT_Gateway/2232_7.md]: Done

**Review by NShaforostov — APPROVED** [review]

**Pull request merged**